### ISSUE TITLE
[Expo 50][Expo AV][Android] NativeEventEmitter called without add and remove listeners

### ISSUE THREAD

**Issue by ajpaulingalls** [expo/expo#26401]
### Minimal reproducible example

https://github.com/ajpaulingalls/bugRepro

### Summary

Using the latest builds of expo and expo-av and trying to play a sound on Android causes the following yellow box:

```
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
```
And the sound does not play.  In debugging the code, it appears in the EventEmitter constructor that, for expo-av, both `nativeModule.__expo_module_name__` && `NativeModules.EXReactNativeEventEmitter` are null, so the temp hack isn't executed, and eventually the native module is not created down in NativeEventEmitter.

### Environment

```  
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.2.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 21.3.0 - /opt/homebrew/bin/node
      Yarn: 1.22.21 - /opt/homebrew/bin/yarn
      npm: 10.2.4 - /opt/homebrew/bin/npm
      Watchman: 2023.12.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
      Android SDK:
        Android NDK: 24.0.8215888
    IDEs:
      Android Studio: 2023.1 AI-231.9392.1.2311.11255304
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.0-preview.10 => 50.0.0-preview.10
      react: 18.2.0 => 18.2.0
      react-native: 0.73.2 => 0.73.2
    Expo Workflow: bare
```

**Comment by ajpaulingalls** [comment]
to build the above package, I basically did the following:
```
pnpm create expo-app --template blank-typescript@beta
pnpm expo add expo-av
copy and pasted the sample code from the Sound docs on expo.dev
added .npmrc for pnpm support
added an mp3 in the required location
pnpm expo run:android
```

**Comment by Haseebshah936** [comment]
@alanjhughes The issue is not yet fixed I am still facing this issue in sdk 50.0.4.

**Comment by Feli87** [comment]
Error persist
"expo": "~50.0.5",
"expo-av": "~13.10.4"
